Lesson4/src/Homework4/Services/MilitaryPart.java

```diff
@@ -0,0 +1,106 @@
+package Homework4.Services;
+
+import Homework4.Model.Person;
+
+import java.util.List;
+import java.util.Scanner;
+
+public class MilitaryPart {
+
+
+    private int size;
+    private Person[] sizeMilitary;
+
+
+    public MilitaryPart(List<Person> fitPeople) {
```

Comment:
Please refactor the constructor.

MilitaryPart should be created without any soldiers, the constructor should only retrieve information about size of the MilitaryPart. You can leave this one too, but then instead of lines 23-32 just do
```
for (Person p : fitPeople) {
    newSoldier(p);
}
```

In that case your application will always add soldier to the MilitaryPart in exactly one way